scripts/utils.sh

```diff
@@ -38,11 +38,11 @@ checkout_branch() {
         BRANCH=main
     fi
     echo "Working from branch $BRANCH"
-    git checkout "${BRANCH}"
     if [ -n "$LOCAL" ]; then
         echo "Working on local changes. There may be uncommitted changes, so skipping upstream pull..."
-    else 
+    else
         git fetch upstream
+        git checkout "${BRANCH}"
         git pull upstream "${BRANCH}"
```

Comment:
Is there any advantage to combining into a single `git checkout -b "${BRANCH}" -t upstream/"${BRANCH}"` ? (perhaps without the `-t`?).

Reply:
I don't tbh, there are mysterious ways on how `git` is working :)